File: LeetCode_ProblemSet/Go/Problem2220.go
package main

import (
	"strconv"
	"strings"
)

func minBitFlips(start int, goal int) int {
    startBin, goalBin := strconv.FormatInt(int64(start), 2), strconv.FormatInt(int64(goal), 2)
	startLen, goalLen := len(startBin), len(goalBin)
	maxLen := max(startLen, goalLen)
	startBin = strings.Join([]string{
		strings.Repeat("0", maxLen - startLen), startBin, 
	}, "")
	goalBin = strings.Join([]string{
		strings.Repeat("0", maxLen - goalLen), goalBin, 
	}, "")
	count := 0
	for i := 0; i < maxLen; i++ {
		if startBin[i] != goalBin[i] {
			count++
		}
	}
	return count
}

File: LeetCode_ProblemSet/Problem2220.py
class Solution(object):
    def minBitFlips(self, start, goal):
        """
        :type start: int
        :type goal: int
        :rtype: int
        """
        start, goal = bin(start)[2:], bin(goal)[2:]
        l = max(len(start), len(goal))
        start, goal = start.rjust(l, "0"), goal.rjust(l, "0")
        return sum([1 for i in range(l) if start[i] != goal[i]])